desktop/main.dev.ts

```diff
@@ -73,52 +73,44 @@ const showCloseAppModal = (parent: BrowserWindow, isDarkMode: boolean): BrowserW
   return child;
 };
 
-const handleClosingApp = async () => {
+let closingApp = false;
```

Comment:
useless flag
app.quit is next line to closingApp = true

Reply:
Not useless, because if it is `false` it prevents from closing the app (`app.on('before-quit', handleClosingApp);`) and waits until the Node process will exit, and then it sets to `true` and calls `app.quit()` again.

I don't like mutable flags, but so far I haven't come up with anything better to make app closing process robust and straightforward (handle closing app in the minimum places: `app.before-quit` and `mainWindow.close` event).
BTW, probably I can use `handleClosingApp` only on `app.before-quit` and handle closing window different, just: `mainWindow.on('close', event => { event.preventDefault(); app.quit(); });` and then it will be handled just in one place.

Reply:
after reading electron docs mainWindow.on('close') is meaningless cause closing single window left will trigger app quit and this will trigger 'before-quit'
also `mainWindow.on('close', event => { event.preventDefault(); app.quit(); });` has the same behaviour as default handler of 'close' event